x-pack/plugins/apm/server/lib/service_map/get_service_map.ts

```diff
@@ -76,21 +75,17 @@ async function getServicesData(options: IEnvOptions) {
     setup: { ...setup, uiFiltersES: [] },
   });
 
-  const { filter } = projection.body.query.bool;
+  const serviceNameFilter = options.serviceName
+    ? [{ term: { [SERVICE_NAME]: options.serviceName } }]
+    : [];
 
-  const params = mergeProjection(projection, {
+  const params = {
```

Comment:
Moving away from `mergeProjection` will make the code a lot more clear - especially for reviewers when things are changed/moved around

Reply:
The reason why we have `mergeProjection` is two-fold:

- `lodash.merge` merges arrays in really weird ways
- if you change a projection, its consumers do not need to update anything (usually).

The downside of spreading only means that you can miss places where the projection is being used. E.g., you'd have to manually inspect all references. Maybe that's a good thing though. Again not a defense - just want to highlight why it's there.

Reply:
> if you change a projection, its consumers do not need to update anything (usually).

This makes sense. Could we achieve the same with unit tests to ensure that projections are correctly applied?

Reply:
that depends on the nature of the change. for additive changes, no. for other changes, yes (arguably already covered by snapshot testing of queries)